airflow/jobs/scheduler_job_runner.py

```diff
@@ -1083,8 +1084,11 @@ def _do_scheduling(self, session: Session) -> int:
             callback_tuples = self._schedule_all_dag_runs(guard, dag_runs, session)
 
         # Send the callbacks after we commit to ensure the context is up to date when it gets run
+        # cache saves time during scheduling of many dag_runs for same dag
+        cached_get_dag: Callable = lru_cache()(lambda dag_id: self.dagbag.get_dag(dag_id, session=session))
```

Comment:
Could benefit from using `functools.partial` instead. Also it’d be a good idea to improve the type hint here so when the callable is called its return value can also be typed instead of being a useless `Any`.

Reply:
I tried to think about how the `functools.partial` could make it better. The proposed statement is a one-liner in adding the `@lru_cache` decorator inline w/o a local additional function declaration. How do you mean it would be better?

In my VSCode at least it shows type hints from parsing python, resulting callable is hinted and the returned `dag` variable in line 1090 is displayed as `(variable) dag: DAG | None` - is it displayed w/o type hints in your IDE?

Reply:
VS Code uses Pylance, which uses additional guessing to infer types. The heuristics mostly work well and are very helpful, but are not always flawless. Correctly annotating the callable would be more friendly for people using other tools.

Mypy identifies the type to be `Any` (using `typing.reveal_type()` to inspect):

```
$ mypy airflow/jobs/scheduler_job_runner.py 
airflow/jobs/scheduler_job_runner.py:1092: note: Revealed type is "Any"
```

Reply:
Thanks for the proposal and thinking a moment about this, it makes the code really "nicer" now. As @AutomationDev85 is off on vacation I'm pushing the update "on behalf".